testing/python/fixtures.py

```diff
@@ -4108,54 +4108,34 @@ def test_fixture_named_request(testdir):
 
 def test_fixture_duplicated_arguments(testdir):
     """Raise error if there are positional and keyword arguments for the same parameter (#1682)."""
-    testdir.makepyfile(
-        """
-        import pytest
+    with pytest.raises(TypeError) as excinfo:
 
-        with pytest.raises(TypeError) as excinfo:
-
-            @pytest.fixture("session", scope="session")
-            def arg(arg):
-                pass
-
-        def test_error():
-            assert (
-                str(excinfo.value)
-                == "The fixture arguments are defined as positional and keyword: scope. "
-                "Use only keyword arguments."
-            )
+        @pytest.fixture("session", scope="session")
+        def arg(arg):
+            pass
 
-    """
+    assert (
+        str(excinfo.value)
+        == "The fixture arguments are defined as positional and keyword: scope. "
+        "Use only keyword arguments."
     )
 
-    reprec = testdir.inline_run()
-    reprec.assertoutcome(passed=1)
-
 
 def test_fixture_with_positionals(testdir):
     """Raise warning, but the positionals should still works (#1682)."""
-    testdir.makepyfile(
-        """
-        import os
+    from _pytest.deprecated import FIXTURE_POSITIONAL_ARGUMENTS
 
-        import pytest
-        from _pytest.deprecated import FIXTURE_POSITIONAL_ARGUMENTS
+    with pytest.warns(pytest.PytestDeprecationWarning) as warnings:
 
-        with pytest.warns(pytest.PytestDeprecationWarning) as warnings:
-            @pytest.fixture("function", [0], True)
-            def arg(monkeypatch):
-                monkeypatch.setenv("AUTOUSE_WORKS", "1")
+        @pytest.fixture("function", [0], True)
+        def arg(monkeypatch):
+            monkeypatch.setenv("AUTOUSE_WORKS", "1")
```

Comment:
Thanks @nicoddemus for improving the tests. Maybe since this fixture isn't actually executed to prove that the `autouse` works, maybe it will be better to remove the unnecessary code from it?
```python
@pytest.fixture("function", [0], True)
def arg():
    pass
```

Reply:
My understanding is that it is desirable to test some other features other than just `autouse`, that's why I kept it. 😁

Reply:
Shall I remove it, or leave as is?

Reply:
Oh sorry, I misunderstood your previous message, yes monkeypatch can be removed, thanks!